### ISSUE TITLE
unsupported Scan, storing driver.Value type []uint8 into type *time.Time

### ISSUE THREAD

**Issue by congjunhua** [go-gorm/gorm#6242]
## Your Question

`Mysql`中的`TIME`类型的字段映射`Go`的`time.Time`类型时报标题错误。

## Expected answer

如何解决？

**Comment by congjunhua** [comment]
将`time.Time`改为`string`可以规避报错，虽然不合理且徒增了类型转化的步骤，但暂时也只能这样了。

望早日修复。

**Comment by black-06** [comment]
You can try `parseTime=True` in your DSN, just like
`gorm:gorm@tcp(localhost:9910)/gorm?charset=utf8&parseTime=True&loc=Local`

**Comment by congjunhua** [comment]
> 

用了，但是不起作用。

**Comment by congjunhua** [comment]
> You can try `parseTime=True` in your DSN, just like `gorm:gorm@tcp(localhost:9910)/gorm?charset=utf8&parseTime=True&loc=Local`

用了，但不起作用。

**Comment by black-06** [comment]
> > You can try `parseTime=True` in your DSN, just like `gorm:gorm@tcp(localhost:9910)/gorm?charset=utf8&parseTime=True&loc=Local`
> 
> 用了，但不起作用。

Could you provide more detailed information

**Comment by BluesFu** [comment]
> You can try `parseTime=True` in your DSN, just like `gorm:gorm@tcp(localhost:9910)/gorm?charset=utf8&parseTime=True&loc=Local`

thanks,  it works well

**Comment by kkk-zxx** [comment]
> provide

parseTime=True.   is support Mysql  date and datetime ,    not support time

**Comment by kkk-zxx** [comment]
> > > You can try `parseTime=True` in your DSN, just like `gorm:gorm@tcp(localhost:9910)/gorm?charset=utf8&parseTime=True&loc=Local`
> > 
> > 
> > 用了，但不起作用。
> 
> Could you provide more detailed information

parseTime=True. is support Mysql date and datetime , not support time

**Comment by black-06** [comment]
> > > > You can try `parseTime=True` in your DSN, just like `gorm:gorm@tcp(localhost:9910)/gorm?charset=utf8&parseTime=True&loc=Local`
> > > 
> > > 
> > > 用了，但不起作用。
> > 
> > 
> > Could you provide more detailed information
> 
> parseTime=True. is support Mysql date and datetime , not support time

Yes, because `github.com/go-sql-driver/mysql` does not support time, see https://github.com/go-sql-driver/mysql/issues/849#issuecomment-416943196 and https://github.com/go-sql-driver/mysql/issues/849#issuecomment-417209270.

If you want to convert MySQL's `Time` to time.Time, this is an example:
```go
type Test struct {
	MyTimeStr string    `gorm:"column:my_time"`
	MyTime    time.Time `gorm:"-"`
	// other fields ...
}

// https://gorm.io/docs/advanced_query.html#Query-Hooks
func (u *Test) AfterFind(tx *gorm.DB) (err error) {
	if u.MyTimeStr != "" {
		u.MyTime, err = time.ParseInLocation("15:04:05", u.MyTimeStr, time.Local)
		if err != nil {
			return err
		}
	}
	return nil
}
```

**Comment by BluesFu** [comment]
https://github.com/golang-module/carbon
